runtime/doc/dev_style.txt

```diff
@@ -41,6 +41,43 @@ All header files should start with `#pragma once` to prevent multiple inclusion.
         #pragma once
 <
 
+Headers system ~
+
+Nvim has uses two types of headers. There are "normal" headers and "defs"
+headers. Typically, each normal header will have a corresponding defs header,
+e.g. fileio.h and fileio_defs.h. This distinction is done to minimize
+recompilation on change. The goal is to achieve the following setup:
+
+All headers (defs and normal) are allowed to include only defs headers, system
+headers and generated declarations. In other words, headers should never
+include normal headers.
+
+.c files can include all headers, normal headers only if they need symbols and
+not types.
+
+Use the following guideline to determine where to put what:
+
+Symbols:
+  - regular function declarations
+  - `extern` variables (including the `EXTERN` macro)
+
+Non-symbols:
+  - macros, i.e. `#define`.
+  - static inline functions, but only if its function declaration has a
+    `REAL_FATTR_ALWAYS_INLINE` attribute.
+  - typedefs
+  - structs
+  - enums
+
+
+• All symbols shall be moved to normal headers.
```

Comment:
```suggestion
- All symbols should be moved to normal headers.
```
Or "must"? (It's _really_ helpful to always stick to the common terminology "can" vs. "should" vs. "must".)

Reply:
I think "must" works here, thanks